cumulusci/salesforce_api/rest_deploy.py

```diff
@@ -98,28 +107,19 @@ def _monitor_deploy_status(self, deploy_request_id):
             response_json = response.json()
 
             if response_json["deployResult"]["status"] != "InProgress":
```

Comment:
I think the logic here is a little bit tricky to follow. Suggestion, if you think it works:

What if we log the state _before_ the conditional (so that the user always sees feedback during the process). Then, make the conditional check for both `InProgress` and `Pending`, so you do not need a condition around the `break`. We exit for any status other than `InProgress` or `Pending`, with the specific handling logic you have for failures.

What do you think?

Reply:
I'm going to go ahead and approve this PR. You can make that change if you agree with it.

Reply:
I understand, I will make these changes first thing tomorrow :)